### PR TITLE
add list functionality

### PR DESCRIPTION
**Pull request by cmokbel1**
Implemented functionality to add a list to the list of lists! Still need to add functionality to re render component upon addition but would like to do this in a seperate branch so that I can create this functionality for both the lists section and the list detail section together.

### CODE REVIEW

File: frontend/src/components/Main.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ export function Main({ userState, setUserState }) {
     }
     return (
         <>
-            <ToDoLists lists={lists} selectedList={selectedList} setSelectedList={setSelectedList} />
+            <ToDoLists lists={lists} selectedList={selectedList} setSelectedList={setSelectedList} addList={addList}/>
```
Comment (1gm) [inline frontend/src/components/Main.js:30]: This code is not formatted

File: frontend/src/components/ToDoLists.js [hunk 2]
```diff
@@ -11,21 +14,48 @@ export const ToDoLists = ({ lists, selectedList, setSelectedList }) => {
...
+            setErrorMessageState('Input Required');
+            return;
+        }
+        if (event.charCode === 13) {
+            const res = await addList(newListName);
+            if (res.error) {
+                setErrorMessageState(res.error);
+            } else {
+                setErrorMessageState('');
+                setMessageState('Post Successful');
```
Comment (1gm) [inline frontend/src/components/ToDoLists.js:33]: 'Post sucessful' doesn't make sense to a user. What does Post mean to a user?

File: frontend/src/components/Main.js (hunk 1 above)
Comment (cmokbel1) [reply frontend/src/components/Main.js:30]: submitting again, not sure if any change happened, format document doesn't seem to have done anything

File: frontend/src/components/ToDoLists.js (hunk 2 above)
Comment (cmokbel1) [reply frontend/src/components/ToDoLists.js]: "List Successfully Added"

**Pull request merged**